OutOfSchool/OutOfSchool.Encryption/Handlers/AppHandlers.cs

```diff
@@ -0,0 +1,55 @@
+using Asp.Versioning.Builder;
+using Microsoft.AspNetCore.Mvc;
+using OutOfSchool.Encryption.Models;
+using OutOfSchool.Encryption.Services;
+
+namespace OutOfSchool.Encryption.Handlers;
+
+public static class AppHandlers
+{
+    public static void Map(WebApplication app, ApiVersionSet apiVersionSet)
+    {
+        app.MapGet(
+                "api/{version:apiVersion}/certificate",
+                ([FromServices] EUSignOAuth2Service euSignOAuth2Service) =>
+                {
+                    var cert = euSignOAuth2Service.GetEnvelopCertificateBase64();
+                    if (cert != null)
+                    {
+                        return Results.Ok(cert);
+                    }
+
+                    return Results.Problem(
+                        title: "Certificate not found",
+                        detail: "Certificate was not present in the system.",
+                        statusCode: 500);
+                })
+            .WithApiVersionSet(apiVersionSet)
+            .MapToApiVersion(1);
+
+        app.MapPost(
+                "api/{version:apiVersion}/decryption", (
+                    EnvelopedUserInfoResponse encryptedUserInfo,
```

Comment:
According to the [docs](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.notnullattribute?view=net-8.0) it is for static analysis only. For service i agree we can add it because we control how the method is called, but for "Controller" method we have no control over how the method will be called in runtime.